domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/AppConfigBasedAmPersistenceSwitch.java

```diff
@@ -0,0 +1,90 @@
+package uk.gov.hmcts.ccd.definition.store.domain;
+
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceReadSource.FROM_AM;
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceReadSource.FROM_CCD;
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceWriteDestination.TO_AM;
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceWriteDestination.TO_BOTH;
+import static uk.gov.hmcts.ccd.definition.store.domain.AmPersistenceWriteDestination.TO_CCD;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.InvalidPropertyException;
+
+import com.google.common.collect.Maps;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+@Named
+@Singleton
+public class AppConfigBasedAmPersistenceSwitch implements AmPersistenceSwitch {
```

Comment:
Can we have a javadoc describing how it works please?

Reply:
Of course, happy to. But the code here is relatively among the simplest ones, where we have a lot more sophisticated logic. If we set the standard for a javadoc inclusion to this level of complexity, then we will have a huge amount of javadoc work by that standard. Here, the code is already documenting itself with long and self-descriptive identifiers/names.